js_modules/dagit/packages/core/src/partitions/SpanRepresentation.tsx

```diff
@@ -0,0 +1,62 @@
+export function assembleIntoSpans<T>(keys: string[], keyTestFn: (key: string, idx: number) => T) {
+  const spans: {startIdx: number; endIdx: number; status: T}[] = [];
+
+  for (let ii = 0; ii < keys.length; ii++) {
+    const status = keyTestFn(keys[ii], ii);
+    if (!spans.length || spans[spans.length - 1].status !== status) {
+      spans.push({startIdx: ii, endIdx: ii, status});
+    } else {
+      spans[spans.length - 1].endIdx = ii;
+    }
+  }
+
+  return spans;
+}
+
+export function stringForSpan(
+  {startIdx, endIdx}: {startIdx: number; endIdx: number},
+  all: string[],
+) {
+  return startIdx === endIdx ? all[startIdx] : `[${all[startIdx]}...${all[endIdx]}]`;
+}
+
+export function allPartitionsSpan({partitionKeys}: {partitionKeys: string[]}) {
+  return stringForSpan({startIdx: 0, endIdx: partitionKeys.length - 1}, partitionKeys);
+}
+
+export function textToPartitions(selected: string, all: string[]) {
+  const terms = selected.split(',').map((s) => s.trim());
+  const result = [];
+  for (const term of terms) {
+    if (term.length === 0) {
+      continue;
+    }
+    const rangeMatch = /^\[(.*)\.\.\.(.*)\]$/g.exec(term);
+    if (rangeMatch) {
+      const [, start, end] = rangeMatch;
+      const allStartIdx = all.indexOf(start);
+      const allEndIdx = all.indexOf(end);
+      if (allStartIdx === -1 || allEndIdx === -1) {
+        throw new Error(`Could not find partitions for provided range: ${start}...${end}`);
+      }
+      result.push(...all.slice(allStartIdx, allEndIdx + 1));
+    } else if (term.includes('*')) {
+      const [prefix, suffix] = term.split('*');
+      result.push(...all.filter((p) => p.startsWith(prefix) && p.endsWith(suffix)));
+    } else {
+      const idx = all.indexOf(term);
+      if (idx === -1) {
+        throw new Error(`Could not find partition: ${term}`);
+      }
+      result.push(term);
+    }
+  }
+  return result.sort((a, b) => all.indexOf(a) - all.indexOf(b));
+}
+
+export function partitionsToText(selected: string[], all: string[]) {
+  return assembleIntoSpans(all, (key) => selected.includes(key))
+    .filter((s) => s.status)
+    .map((s) => stringForSpan(s, all))
+    .join(', ');
+}
```

Comment:
These look like utilities that would be fairly straightforward to write tests for. Mind adding a few?

Reply:
Yep! I think everything in this new file can have tests - will add some today and update this 🙌